debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorTask.java

```diff
@@ -47,7 +48,10 @@ public String version() {
 
     @Override
     public ChangeEventSourceCoordinator<OraclePartition, OracleOffsetContext> start(Configuration config) {
-        OracleConnectorConfig connectorConfig = new OracleConnectorConfig(config);
+        OracleConnectorConfig connectorConfig = new OracleConnectorConfig(
+                config.edit()
+                        .with(AbstractDatabaseHistory.INTERNAL_PREFER_DDL, true)
```

Comment:
As discussed over chat, I don't think that is the right way to go. Rather let's expand the JSON-based history model to convey old and new table name in case of a rename.

Worth thinking a minute about the upgrade/compatibility story, though. Personally, I don't see any problems: that "old table" field would have to be optional anyways, as not all `ALTER` statements do a rename. I.e. the new connector version must be ready to handle events without the value being present. If an old connector version reads back a value _with_ the new field, it may stumble though, which theoretically could happen when downgrading again e.g. in case of another regression, and a rename has been recorded in the history by the new connector version already. OTOH the connector _already_ fails in that situation currently, so this wouldn't make things worse in that regard. It's a good occasion though to make sure we ignore unknown fields during JSON parsing. This will make schema recovery more robust in similar future situations, if not the case yet.

Reply:
An update; I've made the changes related to capturing the previous `TableId` in the alter `TableChange` event.  I'm going to test the forward/backward compatibility between my local branch & main to see if there are any concerns.  

There is one concern I saw that bothers me @gunnarmorling.  In `SchemaChangeEvent` we have a constructor that we delegate to that accepts a `Set<Table>`.  This feels odd when I need to pass in a `TableId` that represents the previous table id that we renamed from.  We don't use this constructor anywhere else so I'm wondering if we shouldn't just adjust this class to return a single `Table` rather than a `Set<Table>` across the board.  wdyt?